pkg/workspace/template_loader.go

```diff
@@ -159,13 +159,20 @@ func (l *TemplateLoader) EvalYAML(libraryCtx LibraryExecutionContext, file *file
 		return nil, nil, err
 	}
 
-	l.ui.Debugf("### ast\n")
-	docSet.Print(l.ui.DebugWriter())
+	l.ui.Debugf("### ast ")
 
 	if !file.IsTemplate() && !file.IsLibrary() || !yamltemplate.HasTemplating(docSet) {
+		docSet.OverrideMapKeys()
```

Comment:
feels a little weird doing it here... but i guess that it makes sense since we are "evaling" yaml. should we leave a comment here?

Reply:
Yeah... I think part of what forces this awkwardness of the location of this logic is that we can't perform this check _until_ the YAML has been parsed. And we don't want to de-dup map keys within a template right here... 🤷🏻 

Yeah, I can contextualize this a bit... good point.